Sources/Rendering/Core/TextActor/index.js

```diff
@@ -0,0 +1,153 @@
+    const fontColor = publicAPI.getProperty().getFontColor();
+    const shadowColor = publicAPI.getProperty().getShadowColor();
+    const shadowOffset = publicAPI.getProperty().getShadowOffset();
+    const shadowBlur = publicAPI.getProperty().getShadowBlur();
+    const resolution = publicAPI.getProperty().getResolution();
+    const backgroundColor = publicAPI.getProperty().getBackgroundColor();
+
+    const dpr = Math.max(window.devicePixelRatio || 1, 1);
+    const canvas = document.createElement('canvas');
+    const ctx = canvas.getContext('2d');
+
+    // Set the text properties to measure
+    const textSize = fontSizeScale(resolution) * dpr;
+
+    ctx.font = `${fontStyle} ${textSize}px "${fontFamily}"`;
+    ctx.textBaseline = 'middle';
+    ctx.textAlign = 'center';
+
+    // Measure the text
+    const metrics = ctx.measureText(text);
+    const textWidth = metrics.width / dpr;
+
+    const {
+      actualBoundingBoxLeft,
+      actualBoundingBoxRight,
+      actualBoundingBoxAscent,
+      actualBoundingBoxDescent,
+    } = metrics;
+    const hAdjustment = (actualBoundingBoxLeft - actualBoundingBoxRight) / 2;
+    const vAdjustment =
+      (actualBoundingBoxAscent - actualBoundingBoxDescent) / 2;
+
+    const textHeight = textSize / dpr - vAdjustment;
+
+    // Update canvas size to fit text
+    const [width, height] = [textWidth * dpr, textHeight * dpr];
+    canvas.width = width;
+    canvas.height = height;
+
+    // vertical flip
+    ctx.translate(0, height);
+    ctx.scale(1, -1);
+
+    // Clear the canvas
+    ctx.clearRect(0, 0, width, height);
+
+    if (backgroundColor) {
+      ctx.fillStyle = vtkMath.floatRGB2HexCode(backgroundColor);
+      ctx.fillRect(0, 0, width, height);
+    }
+
+    // Reset context after resize and prepare for rendering
+    ctx.imageSmoothingEnabled = true;
+    ctx.imageSmoothingQuality = 'high';
+    ctx.font = `${fontStyle} ${textSize}px "${fontFamily}"`;
+    ctx.fillStyle = vtkMath.floatRGB2HexCode(fontColor);
+    ctx.textBaseline = 'middle';
+    ctx.textAlign = 'center';
+
+    // set shadow
+    ctx.shadowColor = shadowColor;
+    ctx.shadowOffsetX = shadowOffset[0];
+    ctx.shadowOffsetY = shadowOffset[1];
+    ctx.shadowBlur = shadowBlur;
+
+    ctx.fillText(text, width / 2 + hAdjustment, height / 2 + vAdjustment);
+
+    const tex = vtkTexture.newInstance();
+    const vtkImage = ImageHelper.canvasToImageData(canvas);
+    tex.setInputData(vtkImage, 0);
+
+    // Update plane dimensions to match text size
+    plane.set({
+      point1: [width, 0, 0],
+      point2: [0, height, 0],
+    });
+
+    return tex;
+  }
+
+  mapper.setInputConnection(plane.getOutputPort());
+
+  publicAPI.setMapper(mapper);
+
+  publicAPI.setInput = (input) => {
+    if (input !== model.input && input) {
+      model.input = input;
+      publicAPI.removeTexture(texture);
+      texture = createTextTexture(model.input);
+      publicAPI.addTexture(texture);
```

Comment:
creating / deleting textures can be expensive. Why don't you reuse the same texture ? FYI, a texture can be "resizable".